src/redis_lock/__init__.py

```diff
@@ -238,15 +239,21 @@ def extend(self, expire=None):
         elif error:
             raise RuntimeError("Unsupported error code %s from EXTEND script" % error)
 
-    def _lock_renewer(self, interval):
+    @staticmethod
+    def _lock_renewer(lockref, interval, stop):
         """
         Renew the lock key in redis every `interval` seconds for as long
         as `self._lock_renewal_thread.should_exit` is False.
         """
         log = getLogger("%s.lock_refresher" % __name__)
-        while not self._lock_renewal_thread.wait_for_exit_request(timeout=interval):
+        while not stop.wait(timeout=interval):
             log.debug("Refreshing lock")
-            self.extend(expire=self._expire)
+            try:
+                lockref().extend(expire=lockref()._expire)
```

Comment:
I'm just not a fan of try/except when there are other ways of handling. :-)

Seen too many bugs causes by overly broad try/except, and this one is overly broad.